src/transaction.ts

```diff
@@ -285,6 +286,7 @@ export class Snapshot extends EventEmitter {
   queryOptions?: IQueryOptions;
   resourceHeader_: {[k: string]: string};
   requestOptions?: Pick<IRequestOptions, 'transactionTag'>;
+  observabilityOptions?: ObservabilityOptions;
```

Comment:
`observabilityOptions` should be injected by customers from [SpannerOptions](https://github.com/googleapis/nodejs-spanner/blob/main/src/index.ts#L134) which should set this property. Do you plan to do that in separate PR ? 
I would recommend to add that in the same PR, so that we can test batch-transaction and mutation flow end to end

Reply:
I shall have it plumbed in progressively as we get there. The reason being that this PR is bound to balloon if I add it straight from the top inside `new Spanner({observabilityOptions: opts})`. Just the way for each part I have been adding it in and testing.